File: bindings/python/examples/how_tos/wallet/check_balance.py
import json
import os

from dotenv import load_dotenv
from iota_sdk import Wallet, WalletOptions

# This example checks the balance of a wallet.

# This example uses secrets in environment variables for simplicity which
# should not be done in production.
load_dotenv()

wallet = Wallet(WalletOptions(storage_path=os.environ.get('WALLET_DB_PATH')))

# Sync wallet with the node
_balance = wallet.sync()

# Just calculate the balance with the known state
balance = wallet.get_balance()
print(f'Balance {json.dumps(balance.to_dict(), indent=4)}')

File: bindings/nodejs/examples/how_tos/wallet/check-balance.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, initLogger } from '@iota/sdk';

// This example uses secrets in environment variables for simplicity which should not be done in production.
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/wallet/check-balance.ts

// This example syncs the account and prints the balance.
async function run() {
    initLogger();
    for (const envVar of ['WALLET_DB_PATH']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }
    try {
        const wallet = await Wallet.create({
            storagePath: process.env.WALLET_DB_PATH,
        });

        // Sync new outputs from the node.
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const _syncBalance = await wallet.sync();

        // After syncing the balance can also be computed with the local data
        const balance = await wallet.getBalance();
        console.log('Balance', balance);
    } catch (error) {
        console.error('Error: ', error);
    }
}

void run().then(() => process.exit());